pkg/crd/gen_integration_test.go

```diff
@@ -119,6 +119,43 @@ var _ = Describe("CRD Generation proper defaulting", func() {
 		expectedOut := string(expectedFileFoos) + string(expectedFileZoos)
 		Expect(out.buf.String()).To(Equal(expectedOut), cmp.Diff(out.buf.String(), expectedOut))
 	})
+
+	It("should add preserveUnknownFields=false when specified", func() {
+		By("calling Generate")
+		no := false
+		gen := &crd.Generator{
+			CRDVersions:           []string{"v1"},
+			PreserveUnknownFields: &no,
+		}
+		Expect(gen.Generate(ctx)).NotTo(HaveOccurred())
+
+		By("searching preserveUnknownFields")
+		Expect(out.buf.String()).To(ContainSubstring("preserveUnknownFields: false"))
+	})
+
+	It("should add preserveUnknownFields=true when specified", func() {
+		By("calling Generate")
+		yes := true
+		gen := &crd.Generator{
+			CRDVersions:           []string{"v1"},
+			PreserveUnknownFields: &yes,
```

Comment:
Shouldn't we disallow adding true for v1?

Reply:
As we don't support generating v1beta1 CRDs anymore, I think it's as easy as "always" disallowing true

Reply:
Is it really necessary? API server does not allow `preserveUnknownFields: true` anyway:
```console
kubectl apply -f pkg/crd/testdata/gen/bar.example.com_foos.yaml
The CustomResourceDefinition "foos.bar.example.com" is invalid: 
* spec.preserveUnknownFields: Invalid value: true: must be false in order to use defaults in the schema
* spec.preserveUnknownFields: Invalid value: true: cannot set to true, set x-kubernetes-preserve-unknown-fields to true in spec.versions[*].schema instead
```

Reply:
I think we could let user specify `true` even though it is not supported but I added a check that prohibits that as suggested.

Reply:
I prefer that we block generation of invalid CRD YAML like in this case. The earlier it fails the better (CRD generation instead of CRD deployment)

Reply:
@sbueringer it is only invalid if you create the CRD as v1. If it was originally created as v1beta1, then it is valid.

Reply:
You can still deploy a v1 CRD with the field set to true if the CRD was before created with v1beta1?

Reply:
> You can still deploy a v1 CRD with the field set to true if the CRD was before created with v1beta1?

Yes

Reply:
Let me know if I should remove the check to allow `true` value.

Reply:
Added a suggestion above to rename the exposed field and prevent possible footguns

Reply:
> Let me know if I should remove the check to allow true value.

Yup, we should allow true. Sorry I thought that v1 CRDs cannot be deployed with true

Reply:
Removed the check to allow `true`.